### DISCUSSION TITLE
Built arm-gcc-embedded for aarch64 - want to contribute a cask

### DISCUSSION THREAD

**Discussion by SeanMollet** [Homebrew/discussions#1156] [Tap maintenance and Homebrew development]
Over the last couple days, I cajoled the arm-none-eabi-gcc toolchain to build on Big Sur/Apple Silicon. I'd like to contribute the cask back, but after digging around, I can't figure out how to mark a cask as M1.

Happy to submit a PR, just need to know exactly what it should look like.

**Comment by carlocab** [comment]
You should probably try to contribute it as a formula to homebrew/core first, instead of as a cask to homebrew/cask, as this looks like a command-line tool.

There are a number of formulae that do
```ruby
depends_on arch: :x86_64
```
so I imagine you should be able to do something similar.

**Reply by SeanMollet** [reply]
This particular bundle of goodies has gone back and forth several times. 
It was a cask and was pulled to become a formula:
https://github.com/Homebrew/homebrew-cask/pull/56802

The formula was deemed to have too delicate/long of a build process:
https://github.com/Homebrew/homebrew-core/pull/45780#issuecomment-569246452

So, it's now a cask again.

I didn't/don't want to re-ignite that debate or get sucked into litigating it. Just want to share a native build for an existing cask.

**Reply by carlocab** [reply]
Is there a binary built for this at https://developer.arm.com? If there is, it might be possible to add it to the existing `gcc-arm-embedded` cask.

If not, then it can't be part of that cask, and you may need to go through the motions of submitting it as a formula anyway. Otherwise, you can host a formula in your own tap (this will allow you to provide pre-built binaries, but you'll probably have to build them on your machine). Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Reply by SeanMollet** [reply]
There isn't.. and getting it to build is non trivial, which is why I want to share it. I don't mind building it, I have to for my own purposes anyway.

Yeah, tap looks good. Thanks.

**Reply by carlocab** [reply]
Sure. There also seem to be a number of related taps linked at https://github.com/Homebrew/homebrew-cask/pull/56802, so you may wish to contribute your formula there.
